### PR TITLE
i18n(fr): add upgrade-to/vX.mdx guides

### PR DESCRIPTION
**Pull request by ArmandPhilippot**
#### Description (required)

* Translate in French [Upgrade to Astro v1](https://docs.astro.build/en/guides/upgrade-to/v1/) (`upgrade-to/v1.mdx`) (including changes made in #8818)
* Translate in French [Upgrade to Astro v2](https://docs.astro.build/en/guides/upgrade-to/v2/) (`upgrade-to/v2.mdx`)
* Translate in French [Upgrade to Astro v3](https://docs.astro.build/en/guides/upgrade-to/v3/) (`upgrade-to/v3.mdx`)
* Translate in French [Upgrade to Astro v4](https://docs.astro.build/en/guides/upgrade-to/v4/) (`upgrade-to/v4.mdx`)
* Fix a typo in `fr/reference/errors/invalid-get-static-paths-entry.mdx` noticed while updating anchors in `v3.mdx`

#### Related issues & labels (optional)

- Suggested label: i18n

#### Additional notes about translation

* I added a PR comment in `upgrade-to/v3.mdx` about `shims` to find out whether you think this word should be translated or not.

* I had doubts on how to translate `Breaking changes` in the different guides, so I leave my reasoning here:
   * `Modifications majeures` could be right but we loose the "breaking" aspect
   * [Electronjs](https://www.electronjs.org/fr/docs/latest/breaking-changes), for example, uses `Changements majeurs avec rupture de compatibilité` but it seems too verbose to repeat it in the sentences
   * I chose to mostly use `Changements cassants`/`Modifications cassantes` and sometimes `majeurs` to avoid repeats
   
In both cases, feel free to leave a suggestion if you have a better idea!

### CODE REVIEW

File: src/content/docs/fr/guides/upgrade-to/v3.mdx [hunk 1]
```diff
@@ -0,0 +1,997 @@
...
+Astro v3.0 supprime entièrement ces API obsolètes. Les [paramètres de configuration](/fr/reference/configuration-reference/) officiellement pris en charge et les syntaxes modernes `<style is:global>` et `<script>` doivent être utilisées à la place.
+
+#### Que devrais-je faire ?
+
+Si vous continuez à utiliser les API v1.x, utilisez plutôt les nouvelles API pour chaque fonctionnalité :
+
+- Options de configuration obsolètes : voir [le guide de migration 0.26](/fr/guides/upgrade-to/v1/#nouvelle-api-de-configuration)
+- Types d'attributs de script/style obsolètes : voir [le guide de migration 0.26](/fr/guides/upgrade-to/v1/#nouveau-comportement-de-script-par-défaut)
+
+### Supprimé : Shims partiels pour les API Web dans le code serveur
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/upgrade-to/v3.mdx:181]: In French, I don't think we have a word for `shims` in this context (computer science). In general context, `shims` could be translated with `cales`. However, in this context (computer science), it seems weird so after checking other French websites, I chose to leave it as `shims` like them.
If you think it should be translated, feel free to leave a suggestion.

File: src/content/docs/fr/guides/upgrade-to/v2.mdx [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+
+ Dans `astro.config.mjs`, ajoutez une barre oblique finale à l'URL définie dans `site`.
+
+```js del={5} ins={6}
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  site: 'https://exemple.com',
+  site: 'https://exemple.com/',
```
Comment (thomasbnt) [inline src/content/docs/fr/guides/upgrade-to/v2.mdx:102]:
```suggestion
export default defineConfig({
  site: 'https://example.com',
  site: 'https://example.com/',
```

File: src/content/docs/fr/guides/upgrade-to/v3.mdx (hunk 1 above)
Comment (thomasbnt) [reply src/content/docs/fr/guides/upgrade-to/v3.mdx:181]: leave this word 👍🏼

**Review by thomasbnt — CHANGES_REQUESTED** [review]
LGTM, good work!

**Review by thomasbnt — APPROVED** [review]
LGTM!

**Review by yanthomasdev — APPROVED** [review]
Damn, this is a big one, LGTM! I hope y'all are ready for the V5 upgrade guide 😆

**Pull request merged**